make/common/native/Link.gmk

```diff
@@ -69,10 +69,15 @@ endef
 ################################################################################
 define CreateStaticLibrary
   # Include partial linking when building the static library with clang on linux
-  ifeq ($(call isTargetOs, linux), true)
-    ifneq ($(findstring $(TOOLCHAIN_TYPE), clang), )
-      $1_ENABLE_PARTIAL_LINKING := true
-    endif
+  ifeq ($(call isTargetOs, linux macosx), true)
```

Comment:
Is this mainly for `clang` support for now?

Reply:
I'm not sure I understand what you mean. I have been continually testing clang on macOS and linux, and gcc on linux, while developing this patch. Since these are the only supported compilers on these platforms, I have a platform-specific test rather than a compiler-specific test.